canonical/opensearch-operator
# Add revision to the dashboard title of the Grafana

With this change, two different OpenSearch applications in different revisions, will have separate dashboards when related to COS.

- data-platform-helpers was added to use the get_charm_revision
- titles of dashboards are changed during the charm upgrade hook
- config_changed was added on refresh_events of COSAgentProvider to re-render the dashboards after a charm upgrade

## How to test it

1. Deploy more than one opensearch application
2. Relate both apps with grafana-agent
3. In one of the applications use juju refresh and apply a local build of this patch
4. Go to the Grafana dashboards an you will see that two dashboards are created:
  - one with no revision (from charmhub)
  - one with Rev 0 (meaning that is a local build)
  
  
 
![image](https://github.com/user-attachments/assets/d9d82ab6-10b3-4d00-bfa2-f864bbad4d9a)
